examples/GroupCharacters.jl

```diff
@@ -0,0 +1,350 @@
+"""
+This is a first attempt to implement group characters in Oscar.
+
+The idea is that the available GAP objects (groups, character tables,
+class functions) are used in a first step, and that access to character
+values yields `QabElem` objects.
+
+Once we agree on the functionality and the integration into Oscar,
+this setup can in a second step be replaced by one that uses
+native Julia objects for representing class functions,
+but character tables and groups still have some counterpart in GAP.
+
+In a third step, we replace the character table objects by native Julia
+objects.
+
+
+Some open questions:
+
+- Would it be advisable to introduce a `ClassFunctionSpace` object?
+  Magma does this, and one benefit would be that this would be a natural
+  object for constructing class functions, via `S(values)`, say.
+  On the other hand, one can also use the character table object itself
+  for this purpose.
+
+- What is the recommended output for a function `character_field`?
+  In general, we get an abelian number field, and one possibility would be
+  to return an embedding into the enveloping cyclotomic field;
+  or would an embedding into `QabField` be more natural?
+  An embedding into the minimal splitting field might be too large
+  for being considered.
+  (And here the question about caching arises, one wants to reuse
+  field objects which have already been created.)
+
+- What is the recommended way to convert those `QabElem` or `nf_elem` values
+  that lie in the base field to integers (or rationals)?
+  For example, it makes sense that `degree(chi)` returns an integer
+  (and not a `Qab_elem`) if `chi` is a (virtual) character.
+
+- What is the recommended way to implement caching in Julia?
+  The most critical point is that a character table of a group
+  relies on a fixed ordering of conjugacy classes.
+  (O.k., this is important enough such that we should provide a field
+  in the table struct hat holds the array of conjugacy classes.)
+
+  The next instance is the connection from the group to its character table.
+  If one wants something like `trivial_character(G)` then the group must
+  know how to fetch the same character table object in each call;
+  perhaps we do not want to support such functions at all.
```

Comment:
I think a group should indeed store "its" character table. The easiest way to do this for now would be to add a `GAPGroupCharacterTable` entry to each concrete GAPGroup subtype (which by default would be left undefined).

More flexible would be to use the `@declare_other` in those types; this adds `Dict{Symbol,Any}` to them, which can be used to store attributes, basically exactly as in GAP component objects. Then one can use `get_special` / `set_special` to access those (see the code in https://github.com/Nemocas/AbstractAlgebra.jl/blob/171944b8c6262d5928316a7bc0ecef41258facf8/src/AbstractAlgebra.jl#L146 and following; see e.g. https://github.com/oscar-system/Oscar.jl/blob/master/src/Modules/FreeModules-graded.jl for a usage example).

This system is still rather bare, and apparently also there are some concerns about its performance; but I think we should just use that now and then refine it as needed.

Reply:
Thanks for the hint. I will use the `@declare_other` approach.